### PR TITLE
Remove `server.listen` and `sails lift` CLI

### PR DESCRIPTION
**Pull request by benbuckman**
For https://www.pivotaltracker.com/story/show/117469133
1. Remove `server.listen` from Sails – the app should handle this instead.
2. Remove the `sails lift` terminal command – no longer useful.

### CODE REVIEW

**Comment by kevinburkeshyp** [comment]
FYI, after this we don't use/care about `liftTimeout`, we should remove it from the API.

**Comment by benbuckman** [comment]
> FYI, after this we don't use/care about `liftTimeout`

Great, removed in https://github.com/Shyp/shyp_api/commit/20d27131d0f803c2af1c08356e7c0959aecf56d7

**Comment by kevinburkeshyp** [comment]
I'm pretty sure we can kill `explicitHost` as well, but don't care about that as much.

File: test/integration/helpers/appHelper.js [hunk 1]
```diff
@@ -104,7 +104,10 @@ module.exports.lift = function(options, callback) {
 	Sails().lift(options, function(err, sails) {
 		if (err) return callback(err);
 		sails.kill = sails.lower;
-		return callback(null, sails);
+
+		sails.hooks.http.server.listen(sails.config.port, function() {
```
Comment (kevinburkeshyp) [inline test/integration/helpers/appHelper.js:108]: should we just do `sails.hooks.http.server.listen(sails.config.port, callback)` ?

**Comment by kevinburkeshyp** [comment]
LGTM

File: test/integration/helpers/appHelper.js (hunk 1 above)
Comment (benbuckman) [inline test/integration/helpers/appHelper.js:108]: No, because `listen`'s callback doesn't have the same `null, sails` signature.

**Pull request merged**